### ISSUE TITLE
[UI/UX] Convert RomM as a PWA

### ISSUE THREAD

**Issue by zurdi15** [rommapp/romm#139]
As many users use RomM from a smartphone, it would be nice to convert RomM as a PWA

**Comment by calebsmithdev** [comment]
This kind of ties into another issue regarding the refactor on components. Is there any interest in adjusting the frontend to use Nuxt, instead of just the base Vue framework? That would streamline the PWA dev quite a bit and simplify a few other pieces as well (like the router removal and more state management options).

If there is any interest here, I can put together a proof of concept to see how it feels/works with the current setup.

**Comment by GAntoine** [comment]
Hey @calebsmithdev, zurdi just merged in a pretty big refactor of the frontend, and things are (IMO) in a much better place right now. If you're still up for it, think you could whip up small Nuxt POC? Ideally something I can use to play with locally, try and get a feel for Nuxt.

FWIW I've used NextJS a little in the past, and I assume the benefits are similar?

**Comment by zurdi15** [comment]
Oh sorry, I wasn't aware or forgot about your "offer" @calebsmithdev. It would be nice if you still want to help with that POC.

In any case, what are the benefits of Nuxt/NextJS?

**Comment by calebsmithdev** [comment]
I was waiting on that refactor for I even tempted to touch it haha. It was looking pretty beefy!

The big benefits for Nuxt here that I see:

- Improved cache handling at the assets level, including rehydration and cache busting
- Dynamic routing. It's all implied routing based on your directory/page structure versus a defined vue-router structure. This one varies in terms of usefulness depending on how explicit you prefer to be.
- Built-in support for PWA without any additional configuration
- Global state is part of the package, so Pinia (and other store alternatives) are less of a requirement across components
- "Automagical" imports for components and composables. Some people love this, some people absolutely hate it.
- It's an opinionated framework that is well documented. Helps maintain a structure at scale.

It had some downsides, specifically around the mindset of knowing when to make something SSR compatible versus client only, but you get used to it pretty quickly. Outside of that - it's essentially just NextJS but for Vue, especially with Nuxt 3.

All that said - sure I can put together a POC and we can see how it feels.

**Comment by calebsmithdev** [comment]
Sorry! Time slipped away from me and I got busy on other projects. Is this something we still want to explore? I saw a V2 PR in draft, so it'd be worthwhile to wait for that for any deep dives, but Nuxt might be overkill at the moment and worth waiting until the UI feels to be in a more stable state.

**Comment by GAntoine** [comment]
>  Is this something we still want to explore? 

It is! But you're right, we should wait until features are more stable before considering a move. At the moment we're still overhauling major parts of the app.
